core/src/main/java/org/apache/calcite/sql/validate/SqlConformance.java

```diff
@@ -102,6 +102,28 @@ public interface SqlConformance {
    */
   boolean isGroupByAlias();
 
+  /**
+   * Value describing how to perform lookup for values defined in a SELECT statement.
+   */
+  enum SelectAliasLookup {
+    /** Values defined in a SELECT statement are not visible within the statement. */
+    UNSUPPORTED,
+    /** Values defined in a SELECT statement are visible to the right of their definition. */
+    LEFT_TO_RIGHT,
```

Comment:
```
What if the table t already has a column called “a” ?
Should “b” be 42 + 1, or should it be the column “a” + 1 ?
```
In Spark:
```
select 2 as x, x + 2 from table;
```
When the table has column x. and value 1. The SQL will return `2   3`.
I'm not sure if any other databases return `2 4` means the priority of column aliases is higher than the definition of the table.

Reply:
As I have commented in the JIRA, if you would do that, you would break all existing queries that redefine columns that exist already. So I don't think any database can afford to do that.

Reply:
If 'x' exists in the FROM, it is used from the FROM. 
Only when lookup in the FROM fails it is made in the local SELECT.